File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/xmlFactory/GenClass/RiskMonitor-0-1/Excel_eventCalcInfo.cpp
// Excel_eventCalcInfo.cpp 
#include "Excel_eventCalcInfo.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Excel_eventCalcInfo::Excel_eventCalcInfo(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //excel_typeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* excel_typeNode = xmlNode->FirstChildElement("excel_type");

   if(excel_typeNode){excel_typeIsNull_ = false;}
   else{excel_typeIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- excel_typeNode , address : " << excel_typeNode << std::endl;
   #endif
   if(excel_typeNode)
   {
       excel_type_ = boost::shared_ptr<XsdTypeToken>(new XsdTypeToken(excel_typeNode));
   }

   //excel_singleRangeEventCalNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* excel_singleRangeEventCalNode = xmlNode->FirstChildElement("excel_singleRangeEventCal");

   if(excel_singleRangeEventCalNode){excel_singleRangeEventCalIsNull_ = false;}
   else{excel_singleRangeEventCalIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- excel_singleRangeEventCalNode , address : " << excel_singleRangeEventCalNode << std::endl;
   #endif
   if(excel_singleRangeEventCalNode)
   {
       excel_singleRangeEventCal_ = boost::shared_ptr<Excel_singleRangeEventCal>(new Excel_singleRangeEventCal(excel_singleRangeEventCalNode));
   }

   //excel_dualRangeEventCalNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* excel_dualRangeEventCalNode = xmlNode->FirstChildElement("excel_dualRangeEventCal");

   if(excel_dualRangeEventCalNode){excel_dualRangeEventCalIsNull_ = false;}
   else{excel_dualRangeEventCalIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- excel_dualRangeEventCalNode , address : " << excel_dualRangeEventCalNode << std::endl;
   #endif
   if(excel_dualRangeEventCalNode)
   {
       excel_dualRangeEventCal_ = boost::shared_ptr<Excel_dualRangeEventCal>(new Excel_dualRangeEventCal(excel_dualRangeEventCalNode));
   }

   //excel_singleBarrierEventCalNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* excel_singleBarrierEventCalNode = xmlNode->FirstChildElement("excel_singleBarrierEventCal");

   if(excel_singleBarrierEventCalNode){excel_singleBarrierEventCalIsNull_ = false;}
   else{excel_singleBarrierEventCalIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- excel_singleBarrierEventCalNode , address : " << excel_singleBarrierEventCalNode << std::endl;
   #endif
   if(excel_singleBarrierEventCalNode)
   {
       excel_singleBarrierEventCal_ = boost::shared_ptr<Excel_singleBarrierEventCal>(new Excel_singleBarrierEventCal(excel_singleBarrierEventCalNode));
   }

   //excel_dualBarrierEventCalNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* excel_dualBarrierEventCalNode = xmlNode->FirstChildElement("excel_dualBarrierEventCal");

   if(excel_dualBarrierEventCalNode){excel_dualBarrierEventCalIsNull_ = false;}
   else{excel_dualBarrierEventCalIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- excel_dualBarrierEventCalNode , address : " << excel_dualBarrierEventCalNode << std::endl;
   #endif
   if(excel_dualBarrierEventCalNode)
   {
       excel_dualBarrierEventCal_ = boost::shared_ptr<Excel_dualBarrierEventCal>(new Excel_dualBarrierEventCal(excel_dualBarrierEventCalNode));
   }

   //excel_identityEventCalNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* excel_identityEventCalNode = xmlNode->FirstChildElement("excel_identityEventCal");

   if(excel_identityEventCalNode){excel_identityEventCalIsNull_ = false;}
   else{excel_identityEventCalIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- excel_identityEventCalNode , address : " << excel_identityEventCalNode << std::endl;
   #endif
   if(excel_identityEventCalNode)
   {
       excel_identityEventCal_ = boost::shared_ptr<Excel_identityEventCal>(new Excel_identityEventCal(excel_identityEventCalNode));
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeToken> Excel_eventCalcInfo::getExcel_type()
{
   if(!this->excel_typeIsNull_){
        return this->excel_type_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeToken>();
   }
}
boost::shared_ptr<Excel_singleRangeEventCal> Excel_eventCalcInfo::getExcel_singleRangeEventCal()
{
   if(!this->excel_singleRangeEventCalIsNull_){
        return this->excel_singleRangeEventCal_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Excel_singleRangeEventCal>();
   }
}
boost::shared_ptr<Excel_dualRangeEventCal> Excel_eventCalcInfo::getExcel_dualRangeEventCal()
{
   if(!this->excel_dualRangeEventCalIsNull_){
        return this->excel_dualRangeEventCal_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Excel_dualRangeEventCal>();
   }
}
boost::shared_ptr<Excel_singleBarrierEventCal> Excel_eventCalcInfo::getExcel_singleBarrierEventCal()
{
   if(!this->excel_singleBarrierEventCalIsNull_){
        return this->excel_singleBarrierEventCal_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Excel_singleBarrierEventCal>();
   }
}
boost::shared_ptr<Excel_dualBarrierEventCal> Excel_eventCalcInfo::getExcel_dualBarrierEventCal()
{
   if(!this->excel_dualBarrierEventCalIsNull_){
        return this->excel_dualBarrierEventCal_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Excel_dualBarrierEventCal>();
   }
}
boost::shared_ptr<Excel_identityEventCal> Excel_eventCalcInfo::getExcel_identityEventCal()
{
   if(!this->excel_identityEventCalIsNull_){
        return this->excel_identityEventCal_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Excel_identityEventCal>();
   }
}
}

File: OTCDerivativesCalculatorModule/Excel_Interface2/GeneratedClass/RiskMonitor-0-1/Excel_eventCalcInfo.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Excel_eventCalcInfo : ISerialized
    {
        public Excel_eventCalcInfo() { }
        public Excel_eventCalcInfo(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode excel_typeNode = xmlNode.SelectSingleNode("excel_type");
            
            if (excel_typeNode != null)
            {
                if (excel_typeNode.Attributes["href"] != null || excel_typeNode.Attributes["id"] != null) 
                {
                    if (excel_typeNode.Attributes["id"] != null) 
                    {
                        excel_typeIDRef_ = excel_typeNode.Attributes["id"].Value;
                        XsdTypeToken ob = new XsdTypeToken(excel_typeNode);
                        IDManager.SetID(excel_typeIDRef_, ob);
                    }
                    else if (excel_typeNode.Attributes["href"] != null)
                    {
                        excel_typeIDRef_ = excel_typeNode.Attributes["href"].Value;
                    }
                    else
                    {
                        excel_type_ = new XsdTypeToken(excel_typeNode);
                    }
                }
                else
                {
                    excel_type_ = new XsdTypeToken(excel_typeNode);
                }
            }
            
        
            XmlNode excel_singleRangeEventCalNode = xmlNode.SelectSingleNode("excel_singleRangeEventCal");
            
            if (excel_singleRangeEventCalNode != null)
            {
                if (excel_singleRangeEventCalNode.Attributes["href"] != null || excel_singleRangeEventCalNode.Attributes["id"] != null) 
                {
                    if (excel_singleRangeEventCalNode.Attributes["id"] != null) 
                    {
                        excel_singleRangeEventCalIDRef_ = excel_singleRangeEventCalNode.Attributes["id"].Value;
                        Excel_singleRangeEventCal ob = new Excel_singleRangeEventCal(excel_singleRangeEventCalNode);
                        IDManager.SetID(excel_singleRangeEventCalIDRef_, ob);
                    }
                    else if (excel_singleRangeEventCalNode.Attributes["href"] != null)
                    {
                        excel_singleRangeEventCalIDRef_ = excel_singleRangeEventCalNode.Attributes["href"].Value;
                    }
                    else
                    {
                        excel_singleRangeEventCal_ = new Excel_singleRangeEventCal(excel_singleRangeEventCalNode);
                    }
                }
                else
                {
                    excel_singleRangeEventCal_ = new Excel_singleRangeEventCal(excel_singleRangeEventCalNode);
                }
            }
            
        
            XmlNode excel_dualRangeEventCalNode = xmlNode.SelectSingleNode("excel_dualRangeEventCal");
            
            if (excel_dualRangeEventCalNode != null)
            {
                if (excel_dualRangeEventCalNode.Attributes["href"] != null || excel_dualRangeEventCalNode.Attributes["id"] != null) 
                {
                    if (excel_dualRangeEventCalNode.Attributes["id"] != null) 
                    {
                        excel_dualRangeEventCalIDRef_ = excel_dualRangeEventCalNode.Attributes["id"].Value;
                        Excel_dualRangeEventCal ob = new Excel_dualRangeEventCal(excel_dualRangeEventCalNode);
                        IDManager.SetID(excel_dualRangeEventCalIDRef_, ob);
                    }
                    else if (excel_dualRangeEventCalNode.Attributes["href"] != null)
                    {
                        excel_dualRangeEventCalIDRef_ = excel_dualRangeEventCalNode.Attributes["href"].Value;
                    }
                    else
                    {
                        excel_dualRangeEventCal_ = new Excel_dualRangeEventCal(excel_dualRangeEventCalNode);
                    }
                }
                else
                {
                    excel_dualRangeEventCal_ = new Excel_dualRangeEventCal(excel_dualRangeEventCalNode);
                }
            }
            
        
            XmlNode excel_singleBarrierEventCalNode = xmlNode.SelectSingleNode("excel_singleBarrierEventCal");
            
            if (excel_singleBarrierEventCalNode != null)
            {
                if (excel_singleBarrierEventCalNode.Attributes["href"] != null || excel_singleBarrierEventCalNode.Attributes["id"] != null) 
                {
                    if (excel_singleBarrierEventCalNode.Attributes["id"] != null) 
                    {
                        excel_singleBarrierEventCalIDRef_ = excel_singleBarrierEventCalNode.Attributes["id"].Value;
                        Excel_singleBarrierEventCal ob = new Excel_singleBarrierEventCal(excel_singleBarrierEventCalNode);
                        IDManager.SetID(excel_singleBarrierEventCalIDRef_, ob);
                    }
                    else if (excel_singleBarrierEventCalNode.Attributes["href"] != null)
                    {
                        excel_singleBarrierEventCalIDRef_ = excel_singleBarrierEventCalNode.Attributes["href"].Value;
                    }
                    else
                    {
                        excel_singleBarrierEventCal_ = new Excel_singleBarrierEventCal(excel_singleBarrierEventCalNode);
                    }
                }
                else
                {
                    excel_singleBarrierEventCal_ = new Excel_singleBarrierEventCal(excel_singleBarrierEventCalNode);
                }
            }
            
        
            XmlNode excel_dualBarrierEventCalNode = xmlNode.SelectSingleNode("excel_dualBarrierEventCal");
            
            if (excel_dualBarrierEventCalNode != null)
            {
                if (excel_dualBarrierEventCalNode.Attributes["href"] != null || excel_dualBarrierEventCalNode.Attributes["id"] != null) 
                {
                    if (excel_dualBarrierEventCalNode.Attributes["id"] != null) 
                    {
                        excel_dualBarrierEventCalIDRef_ = excel_dualBarrierEventCalNode.Attributes["id"].Value;
                        Excel_dualBarrierEventCal ob = new Excel_dualBarrierEventCal(excel_dualBarrierEventCalNode);
                        IDManager.SetID(excel_dualBarrierEventCalIDRef_, ob);
                    }
                    else if (excel_dualBarrierEventCalNode.Attributes["href"] != null)
                    {
                        excel_dualBarrierEventCalIDRef_ = excel_dualBarrierEventCalNode.Attributes["href"].Value;
                    }
                    else
                    {
                        excel_dualBarrierEventCal_ = new Excel_dualBarrierEventCal(excel_dualBarrierEventCalNode);
                    }
                }
                else
                {
                    excel_dualBarrierEventCal_ = new Excel_dualBarrierEventCal(excel_dualBarrierEventCalNode);
                }
            }
            
        
            XmlNode excel_identityEventCalNode = xmlNode.SelectSingleNode("excel_identityEventCal");
            
            if (excel_identityEventCalNode != null)
            {
                if (excel_identityEventCalNode.Attributes["href"] != null || excel_identityEventCalNode.Attributes["id"] != null) 
                {
                    if (excel_identityEventCalNode.Attributes["id"] != null) 
                    {
                        excel_identityEventCalIDRef_ = excel_identityEventCalNode.Attributes["id"].Value;
                        Excel_identityEventCal ob = new Excel_identityEventCal(excel_identityEventCalNode);
                        IDManager.SetID(excel_identityEventCalIDRef_, ob);
                    }
                    else if (excel_identityEventCalNode.Attributes["href"] != null)
                    {
                        excel_identityEventCalIDRef_ = excel_identityEventCalNode.Attributes["href"].Value;
                    }
                    else
                    {
                        excel_identityEventCal_ = new Excel_identityEventCal(excel_identityEventCalNode);
                    }
                }
                else
                {
                    excel_identityEventCal_ = new Excel_identityEventCal(excel_identityEventCalNode);
                }
            }
            
        
        }
        
    
        #region Excel_type_
        private XsdTypeToken excel_type_;
        public XsdTypeToken Excel_type_
        {
            get
            {
                if (this.excel_type_ != null)
                {
                    return this.excel_type_; 
                }
                else if (this.excel_typeIDRef_ != null)
                {
                    excel_type_ = IDManager.getID(excel_typeIDRef_) as XsdTypeToken;
                    return this.excel_type_; 
                }
                else
                {
                    throw new Exception( "excel_type_Node no exist!");
                }
            }
            set
            {
                if (this.excel_type_ != value)
                {
                    this.excel_type_ = value;
                }
            }
        }
        #endregion
        
        public string excel_typeIDRef_ { get; set; }
        #region Excel_singleRangeEventCal_
        private Excel_singleRangeEventCal excel_singleRangeEventCal_;
        public Excel_singleRangeEventCal Excel_singleRangeEventCal_
        {
            get
            {
                if (this.excel_singleRangeEventCal_ != null)
                {
                    return this.excel_singleRangeEventCal_; 
                }
                else if (this.excel_singleRangeEventCalIDRef_ != null)
                {
                    excel_singleRangeEventCal_ = IDManager.getID(excel_singleRangeEventCalIDRef_) as Excel_singleRangeEventCal;
                    return this.excel_singleRangeEventCal_; 
                }
                else
                {
                    throw new Exception( "excel_singleRangeEventCal_Node no exist!");
                }
            }
            set
            {
                if (this.excel_singleRangeEventCal_ != value)
                {
                    this.excel_singleRangeEventCal_ = value;
                }
            }
        }
        #endregion
        
        public string excel_singleRangeEventCalIDRef_ { get; set; }
        #region Excel_dualRangeEventCal_
        private Excel_dualRangeEventCal excel_dualRangeEventCal_;
        public Excel_dualRangeEventCal Excel_dualRangeEventCal_
        {
            get
            {
                if (this.excel_dualRangeEventCal_ != null)
                {
                    return this.excel_dualRangeEventCal_; 
                }
                else if (this.excel_dualRangeEventCalIDRef_ != null)
                {
                    excel_dualRangeEventCal_ = IDManager.getID(excel_dualRangeEventCalIDRef_) as Excel_dualRangeEventCal;
                    return this.excel_dualRangeEventCal_; 
                }
                else
                {
                    throw new Exception( "excel_dualRangeEventCal_Node no exist!");
                }
            }
            set
            {
                if (this.excel_dualRangeEventCal_ != value)
                {
                    this.excel_dualRangeEventCal_ = value;
                }
            }
        }
        #endregion
        
        public string excel_dualRangeEventCalIDRef_ { get; set; }
        #region Excel_singleBarrierEventCal_
        private Excel_singleBarrierEventCal excel_singleBarrierEventCal_;
        public Excel_singleBarrierEventCal Excel_singleBarrierEventCal_
        {
            get
            {
                if (this.excel_singleBarrierEventCal_ != null)
                {
                    return this.excel_singleBarrierEventCal_; 
                }
                else if (this.excel_singleBarrierEventCalIDRef_ != null)
                {
                    excel_singleBarrierEventCal_ = IDManager.getID(excel_singleBarrierEventCalIDRef_) as Excel_singleBarrierEventCal;
                    return this.excel_singleBarrierEventCal_; 
                }
                else
                {
                    throw new Exception( "excel_singleBarrierEventCal_Node no exist!");
                }
            }
            set
            {
                if (this.excel_singleBarrierEventCal_ != value)
                {
                    this.excel_singleBarrierEventCal_ = value;
                }
            }
        }
        #endregion
        
        public string excel_singleBarrierEventCalIDRef_ { get; set; }
        #region Excel_dualBarrierEventCal_
        private Excel_dualBarrierEventCal excel_dualBarrierEventCal_;
        public Excel_dualBarrierEventCal Excel_dualBarrierEventCal_
        {
            get
            {
                if (this.excel_dualBarrierEventCal_ != null)
                {
                    return this.excel_dualBarrierEventCal_; 
                }
                else if (this.excel_dualBarrierEventCalIDRef_ != null)
                {
                    excel_dualBarrierEventCal_ = IDManager.getID(excel_dualBarrierEventCalIDRef_) as Excel_dualBarrierEventCal;
                    return this.excel_dualBarrierEventCal_; 
                }
                else
                {
                    throw new Exception( "excel_dualBarrierEventCal_Node no exist!");
                }
            }
            set
            {
                if (this.excel_dualBarrierEventCal_ != value)
                {
                    this.excel_dualBarrierEventCal_ = value;
                }
            }
        }
        #endregion
        
        public string excel_dualBarrierEventCalIDRef_ { get; set; }
        #region Excel_identityEventCal_
        private Excel_identityEventCal excel_identityEventCal_;
        public Excel_identityEventCal Excel_identityEventCal_
        {
            get
            {
                if (this.excel_identityEventCal_ != null)
                {
                    return this.excel_identityEventCal_; 
                }
                else if (this.excel_identityEventCalIDRef_ != null)
                {
                    excel_identityEventCal_ = IDManager.getID(excel_identityEventCalIDRef_) as Excel_identityEventCal;
                    return this.excel_identityEventCal_; 
                }
                else
                {
                    throw new Exception( "excel_identityEventCal_Node no exist!");
                }
            }
            set
            {
                if (this.excel_identityEventCal_ != value)
                {
                    this.excel_identityEventCal_ = value;
                }
            }
        }
        #endregion
        
        public string excel_identityEventCalIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

